### ISSUE TITLE
React 18 compatibility

### ISSUE THREAD

**Issue by onionhammer** [s-yadav/react-number-format#629]
**Description/Screenshot**
```
npm WARN Could not resolve dependency:
npm WARN peer react@"^0.14 || ^15.0.0 || ^16.0.0 || ^17.0.0" from react-number-format@4.9.1      
npm WARN node_modules/react-number-format
npm WARN   react-number-format@"^4.9.1" from the root project
npm WARN 
npm WARN Conflicting peer dependency: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@"^0.14 || ^15.0.0 || ^16.0.0 || ^17.0.0" from react-number-format@4.9.1    
npm WARN   node_modules/react-number-format
npm WARN     react-number-format@"^4.9.1" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-number-format@4.9.1
npm WARN Found: react-dom@18.0.0
npm WARN node_modules/react-dom
npm WARN   react-dom@"^18.0.0" from the root project
npm WARN   2 more (@testing-library/react, use-error-boundary)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer react-dom@"^0.14 || ^15.0.0 || ^16.0.0 || ^17.0.0" from react-number-format@4.9.1  
npm WARN node_modules/react-number-format
npm WARN   react-number-format@"^4.9.1" from the root project
npm WARN 
npm WARN Conflicting peer dependency: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@"^0.14 || ^15.0.0 || ^16.0.0 || ^17.0.0" from react-number-format@4.9.1
npm WARN   node_modules/react-number-format
npm WARN     react-number-format@"^4.9.1" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: react-query@3.34.19
npm WARN Found: react@18.0.0
npm WARN node_modules/react
npm WARN   react@"^18.0.0" from the root project
npm WARN   6 more (@testing-library/react, react-dom, @compiled/react, ...)
```

**Steps to Reproduce**
1. Install latest version of react
2. Install react-number-format@4.9.1

**Comment by kamzata** [comment]
I receive just an ERR not a WARN:

```
❯ npm i react-number-format
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: crea-ricevute@0.1.0
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^0.14 || ^15.0.0 || ^16.0.0 || ^17.0.0" from react-number-format@4.9.1
npm ERR! node_modules/react-number-format
npm ERR!   react-number-format@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
```

**Comment by maxbailey** [comment]
Also having OP's same issue! 🙂

**Comment by s-yadav** [comment]
Released this on 4.9.2 with spec fixes.

**Comment by nikhil-varma** [comment]
Closing this issue in favor of the about comment. Please feel free to comment here for any queries
